0001-typedb/02-dev/06-infer.md

```diff
@@ -0,0 +1,336 @@
+   the query exhaustively. If you only need 1 answer, adding `limit 1;` to the match query can significantly improve 
+   query times.
+2. Using the same transaction for multiple reasoning queries. Because inferred data is cleared between transactions, 
+   running the same or similar queries within one transaction can reuse previously inferred data. Combined with a limit 
+   on the query, it might be possible to avoid having to do any new reasoning at all.
+3. For complex queries, it can also be beneficial to add more CPU cores, as the reasoning engine is able to explore 
+   more paths in the database concurrently.
+
+## How to use inference
+
+Inference is enabled as a read transaction option.
+
+### TypeDB Studio
+
+To send a request with inference enabled in [TypeDB Studio](#client/studio) follow these steps:
+
+1. Ensure the [**Session type**] (schema / data) switch is set to “**data**” (next to the database dropdown).
+2. Ensure the [**Transaction type**] (write / read) switch is set to “**read**”.
+3. Ensure the [**infer**] button (next to the transaction type switch) is pushed (shown in **green** text).
+4. Click the `+` icon at the top of the **Text-editor** panel.
+5. Copy the TypeQL query into the **Text-editor** panel.
+6. Click the green “play” button.
+
+### TypeDB Console
+
+Set the `infer` option to `true` when you start the read transaction. For example:
+
+<!-- test-ignore -->
+```
+transaction typedb data read --infer true
+```
+
+### TypeDB Drivers
+
+Set the `infer` option to `True` when you start the read transaction. For example:
+
+<!-- #todo Consider adding other languages--->
+
+<!-- test-ignore -->
+```python
+typedb_options = TypeDBOptions.core()  # Initialising a new set of options
+typedb_options.infer = True  # Enabling inference in this new set of options
+with session.transaction(TransactionType.READ, typedb_options) as transaction:
+```
+
+## Example
+
+In the IAM database there are multiple rules, but we will use the one described at the 
+[IAM schema explanation](../01-start/04-iam-schema.md) page. 
+
+<!-- test-ignore -->
+```typeql
+rule add-view-access:
+    when {
+        $modify isa action, has action-name "modify_file";
+        $view isa action, has action-name "view_file";
+        $ac_modify (accessed-object: $obj, valid-action: $modify) isa access;
+        $ac_view (accessed-object: $obj, valid-action: $view) isa access;
+        (permitted-subject: $subj, permitted-access: $ac_modify) isa permission;
+    } then {
+        (permitted-subject: $subj, permitted-access: $ac_view) isa permission;
+    };
+```
+
+By default, the IAM dataset used in the [Quickstart guide](../01-start/03-quickstart.md) sets only permissions for the 
+`modify_file` action. 
+
+But the IAM schema has the rule called `add-view-access` (see above) to infer `view_file` action access from the 
+`modify_file` access. The `view_file` action access permission does not exist in the database but can be inferred by 
+the rule.
```

Comment:
@james-whiteside 
This rule infers that the `$subj` has permission to the access, doesn't it? Do you think the wording can be improved.

Reply:
Yeah that all makes sense. "Permission to `view file`" is good I think. I personally think the words "action" and "access" aren't necessary, especially if users will have spent a while with the schema already in earlier examples.

Reply:
Fixed in https://github.com/vaticle/docs/pull/676/commits/205a6e66ac8d8424100a645d8279401839e3b5bb